test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -14010,6 +14010,21 @@ BOOST_AUTO_TEST_CASE(test_underscore_in_hex)
 	ABI_CHECK(callContractFunction("f(bool)", false), encodeArgs(u256(0x1234abcd1234)));
 }
 
+BOOST_AUTO_TEST_CASE(flipping_sign_tests)
+{
+	char const* sourceCode = R"(
+		contract test {
+			function f() public returns (bool){
+				int x = -2**255;
+				assert(-x == x);
+				return true;
+			}
+		}
+	)";
+	compileAndRun(sourceCode);
+	ABI_CHECK(callContractFunction("f()"), encodeArgs(u256(0)));
```

Comment:
The function returns `true`, so this should be `encodeArgs(true)`. Apart from that, perfect!

Reply:
Thanks for the feedback! I did a rebase though, not sure if that's the proper procedure here or if I should've just not updated and pushed the fix anyways.